### ISSUE TITLE
`s!` string formats syntax highlight

### ISSUE THREAD

**Issue by Trebor-Huang** [leanprover/vscode-lean4#274]
I think this should do it? Just stick it in front of the string part:
```json
{
  "begin": "s!\"", "end": "\"",
  "name": ??
  "patterns": [
    { "begin": "\\{", "end": "\\}", "patterns": [ {"include": "$self"} ] },
    { "match": "\\\\[\\\\\"ntr']", "name": "constant.character.escape.lean4" },
    { "match": "\\\\x[0-9A-Fa-f][0-9A-Fa-f]", "name": "constant.character.escape.lean4" },
    { "match": "\\\\u[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]", "name": "constant.character.escape.lean4" }
  ]
},
```
By the way, regexes stored as strings are *terrible*.

**Comment by Trebor-Huang** [comment]
Also, when I input `s!"`, the second quote doesn't appear. I think that's an easy change too?

**Comment by thorimur** [comment]
The smaller, second part of this (making the second quote appear when typing `s!"`) is addressed as part of #330 . :)
